statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,245 @@
+
+        Parameters
+        ----------
+        data : array-like
+          The raw data
+        sampling_method : one of cluster, strata, or prob_weight
+
+        Returns
+        -------
+        A nx1 numpy array
+        """
+
+        # check if str, num, or list
+        if isinstance(sampling_method, str):
+            temp = data[sampling_method].values
+            del data[sampling_method]
+            sampling_method = temp
+            n = len(sampling_method)
+            sampling_method = sampling_method.reshape(n,1)
+        elif isinstance(sampling_method, int):
+            try:
+                temp = data.iloc[:, sampling_method].values
+                data.drop(data.columns[sampling_method], inplace=True, axis=1)
+                sampling_method = temp
+                n = len(sampling_method)
+                sampling_method = sampling_method.reshape(n,1)
+            except AttributeError: # data is a ndarray
+                temp = data[:,sampling_method]
+                data = np.delete(data, sampling_method, 1)
+                sampling_method = temp
+                n = len(sampling_method)
+                sampling_method = sampling_method.reshape(n,1)
+        elif isinstance(sampling_method, list):
+            sampling_method = np.array(sampling_method)
+        elif sampling_method is None:
+            n = data.shape[0]
+            self.sampling_method = np.ones([n,1])
+        else:
+            n = len(sampling_method)
+            sampling_method = sampling_method.reshape(n,1)
+
+        return data, sampling_method
+
+
+    def _create_subgroup_labels(self):
+        """
+        creates labels for each SSU within a PSU so that one is sure that 
+        operations are being done withing each sampling subgroup
+
+        Parameters
+        ----------
+        self
+
+        Returns
+        -------
+        A nx3 array, the first column being self.strata, the second column
+        is self.cluster, the third column is the unique label for each subgroup
+        """
+
+        _, i1 = np.unique(strata, return_inverse=True)
+        _, i2 = np.unique(clusters, return_inverse=True)
+        labels = i1 * len(i2) + i2
+        return labels
+
+    def _jackknife(self, method, column_index):
+        unique_strata = np.unique(self.strata)
+        stratum_stats = np.empty(len(unique_strata))
+        for id, stratum in enumerate(unique_strata):
+            # get indices for a particular stratum and the # of clusters in it
+            id_stratum = np.where(self.strata == stratum)[0]        
+            unique_clusters = np.unique(self.cluster[id_stratum])
+            num_clusters = len(unique_clusters)
+
+            # get the statistic without a particular cluster
+            cluster_stats = np.empty(num_clusters)
+
+            for ind, cluster in enumerate(unique_clusters):
+                new_weights = self._reweight(cluster, id_stratum, method)
+                if method == "total":
```

Comment:
I'm still not sure about weight nomenclature.  This class should not assume that weights need to sum to 1.  Does that mean we have freq_weights instead of prob_weights??